### PR TITLE
[744] Change international degree country input to use country autocomplete

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/pT77mEFm/744-spike-international-degree-country-input-should-use-country-autocomplete

### Changes proposed in this pull request
- Introduce new component: `FormComponents::CountryAutocomplete::View`
- Install another node module: https://github.com/alphagov/govuk-country-and-territory-autocomplete
- Use a custom file (`location-autocomplete-graph.json`) for `govuk-country-and-territory-autocomplete` because DTTP doesn't have all the countries listed in the original [location-autocomplete-graph.json](https://github.com/alphagov/govuk-country-and-territory-autocomplete/blob/master/dist/location-autocomplete-graph.json).
- Add country codes for `Dttp::CodeSets::Countries::MAPPING` to generate a custom `location-autocomplete-graph.json` file
- Rake task to regenerate the file `location-autocomplete-graph.json` in case we add or remove entries in `Dttp::CodeSets::Countries::MAPPING`

### Guidance to review
- Visit http://0.0.0.0:3000/trainees/{trainee_id}/degrees/new?locale_code=non_uk
- Under "Where is the degree institution based?", start typing. e.g. "Wales", "Scotland", etc. You can also try searching for a country using a country code
- You should always find the country you expected to find

### CODE REVIEW

**Comment by jhenderson** [comment]
[[SPIKE] International degree country input should use country autocomplete](https://trello.com/c/pT77mEFm/744-spike-international-degree-country-input-should-use-country-autocomplete)

**Review by AbigailMcP — APPROVED** [review]
Code looks good to me!

Am I right in thinking the addition of `country_code` to `lib/dttp/code_sets/countries.rb` was purely to generate the custom json?

It probably makes sense for them to stick around in the code in case we need to update the json, but I'd be tempted to add a comment to the top of `lib/dttp/code_sets/countries.rb` explaining why `country_code` is included as it's not obvious.

**Comment by jhenderson** [comment]
@AbigailMcP I added the comment to `Dttp::CodeSets::Countries::MAPPING` and also created a rake task to regenerate the file `location-autocomplete-graph.json` in case we add or remove entries in Dttp::CodeSets::Countries::MAPPING.

File: lib/tasks/country_autocomplete_graph.rake [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    (original_location_graph.keys - new_location_graph.keys).each do |primary_key|
+      missing_from_dttp[primary_key] = original_location_graph[primary_key]
+    end
+
+    File.open(Rails.root.join("public", graph_file_name), "w") do |f|
+      f.write(new_location_graph.to_json)
+    end
+
+    File.open(Rails.root.join("missing-location-graph-from-dttp.json"), "w") do |f|
+      f.write(new_location_graph.to_json)
```
Comment (AbigailMcP) [inline lib/tasks/country_autocomplete_graph.rake:39]: I think you mean this to be `missing_from_dttp`

**Review by AbigailMcP — COMMENTED** [review]
Nice! Just one comment on the rake task.

File: lib/tasks/country_autocomplete_graph.rake (hunk 1 above)
Comment (jhenderson) [reply lib/tasks/country_autocomplete_graph.rake]: Good spot. 👍

**Comment by edwardhorsford** [comment]
Looking good!

**Pull request merged**